src/akkudoktoreos/devices/battery.py

```diff
@@ -10,345 +10,271 @@
     )
     discharge_array: list[int] = Field(
-        description="Indicates for each hour whether the EV is discharging (`0` for no discharging, `1` for discharging)."
+        description="Hourly discharging status (0 for no discharging, 1 for discharging)."
     )
-    entlade_effizienz: float = Field(description="The discharge efficiency as a float.")
-    hours: int = Field(description="Amount of hours the simulation is done for.")
-    kapazitaet_wh: int = Field(description="The capacity of the EV’s battery in watt-hours.")
-    lade_effizienz: float = Field(description="The charging efficiency as a float.")
-    max_ladeleistung_w: int = Field(description="The maximum charging power of the EV in watts.")
+    discharge_efficiency: float = Field(description="The discharge efficiency.")
+    hours: int = Field(description="Number of hours in the simulation.")
+    capacity_wh: int = Field(description="Capacity of the EV’s battery in watt-hours.")
+    charge_efficiency: float = Field(description="Charging efficiency.")
+    max_charging_power_w: int = Field(description="Maximum charging power in watts.")
     soc_wh: float = Field(
-        description="The state of charge of the battery in watt-hours at the start of the simulation."
+        description="State of charge of the battery in watt-hours at the start of the simulation."
     )
-    start_soc_prozent: int = Field(
-        description="The state of charge of the battery in percentage at the start of the simulation."
+    initial_soc_percentage: int = Field(
+        description="State of charge at the start of the simulation in percentage."
     )
 
-    @field_validator(
-        "discharge_array",
-        "charge_array",
-        mode="before",
-    )
+    @field_validator("discharge_array", "charge_array", mode="before")
     def convert_numpy(cls, field: Any) -> Any:
         return NumpyEncoder.convert_numpy(field)[0]
 
 
-class PVAkku(DeviceBase):
+class Battery(DeviceBase):
+    """Represents a solar panel battery device with methods to simulate energy charging and discharging."""
+
     def __init__(
         self,
-        parameters: Optional[BaseAkkuParameters] = None,
+        parameters: Optional[BaseBatteryParameters] = None,
         hours: Optional[int] = 24,
         provider_id: Optional[str] = None,
     ):
-        # Configuration initialisation
+        # Initialize configuration and parameters
         self.provider_id = provider_id
         self.prefix = "<invalid>"
         if self.provider_id == "GenericBattery":
             self.prefix = "battery"
         elif self.provider_id == "GenericBEV":
             self.prefix = "bev"
-        # Parameter initialisiation
+
         self.parameters = parameters
-        if hours is None:
-            self.hours = self.total_hours
-        else:
-            self.hours = hours
+        self.hours = hours if hours is not None else 24
```

Comment:
should probably get rid of the hours parameter in the init and use the config self.config. ... value

Reply:
or maybe not, however we should replace the magic 24 by it